### PR TITLE
feat: add raw transactions endpoint

### PR DESCRIPTION
**Pull request by greged93**
Time spent on this PR: 0.3 day

Resolves: #890

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
Adds debug raw transactions.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: tests/debug_api.rs [hunk 1]
```diff
@@ -191,6 +192,104 @@ async fn test_raw_transaction(#[future] katana: Katana, _setup: ()) {
...
+#[rstest]
+#[awt]
+#[tokio::test(flavor = "multi_thread")]
+/// Test for fetching raw transactions by block hash and block number.
+async fn test_raw_transactions(#[future] katana: Katana, _setup: ()) {
+    // Start the Kakarot RPC server.
+    let (server_addr, server_handle) =
+        start_kakarot_rpc_server(&katana).await.expect("Error setting up Kakarot RPC server");
+
+    // Fetch raw receipts by block hash.
```
Comment (Eikix) [inline tests/debug_api.rs:204]: wrong comment

File: tests/debug_api.rs [hunk 2]
```diff
@@ -191,6 +192,104 @@ async fn test_raw_transaction(#[future] katana: Katana, _setup: ()) {
...
+    let raw_by_block_number: Value =
+        serde_json::from_str(&response_by_block_number).expect("Failed to deserialize response body");
+
+    let rlp_bytes_by_block_number: Vec<Bytes> =
+        serde_json::from_value(raw_by_block_number["result"].clone()).expect("Failed to deserialize result");
+
+    // Assert equality of receipts fetched by block hash and block number.
+    assert_eq!(rlp_bytes_by_block_number, rlp_bytes_by_block_hash);
+
+    // Get eth provider
```
Comment (Eikix) [inline tests/debug_api.rs:258]: nit: useless comment

File: tests/debug_api.rs [hunk 3]
```diff
@@ -191,6 +192,104 @@ async fn test_raw_transaction(#[future] katana: Katana, _setup: ()) {
...
+                r: signature.r,
+                s: signature.s,
+                odd_y_parity: signature.y_parity.unwrap_or(reth_rpc_types::Parity(false)).0,
+            },
+        )
+        .envelope_encoded();
+
+        // Assert the equality of the constructed receipt with the corresponding receipt from both block hash and block number.
+        assert_eq!(rlp_bytes_by_block_number[i], rlp_bytes);
+        assert_eq!(rlp_bytes_by_block_hash[i], rlp_bytes);
```
[the remark this answers is not in the document]
Comment (greged93) [reply tests/debug_api.rs:286]: I think it's ok, we are pulling the actual tx and encoding, don't think we need the other way

[the remark this answers is not in the document]
Comment (Eikix) [reply tests/debug_api.rs:285]: we started doing it for the other routes iirc

**Review by Eikix — APPROVED** [review]
lgtm

**Pull request merged**